bundles/org.openhab.binding.windcentrale/src/main/java/org/openhab/binding/windcentrale/internal/handler/WindcentraleWindmillHandler.java

```diff
@@ -0,0 +1,197 @@
+ * sent to one of the channels.
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ * @author Wouter Born - Add null annotations
+ * @author Wouter Born - Add support for new API with authentication
+ */
+@NonNullByDefault
+public class WindcentraleWindmillHandler extends BaseThingHandler {
+
+    private static final long CACHE_EXPIRY = TimeUnit.SECONDS.toMillis(5);
+
+    private final Logger logger = LoggerFactory.getLogger(WindcentraleWindmillHandler.class);
+
+    private @NonNullByDefault({}) WindmillConfiguration config;
+    private @Nullable Windmill windmill;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private final ExpiringCache<@Nullable WindmillStatus> statusCache = new ExpiringCache<>(CACHE_EXPIRY, () -> {
+        try {
+            WindcentraleAPI api = getAPI();
+            Windmill windmill = this.windmill;
+            return api == null || windmill == null ? null : api.getLiveData(windmill);
+        } catch (FailedGettingDataException | InvalidAccessTokenException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            return null;
+        }
+    });
+
+    public WindcentraleWindmillHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Windcentrale handler '{}'", getThing().getUID());
+        final ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+    }
+
+    protected @Nullable WindcentraleAPI getAPI() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            return null;
+        }
+        WindcentraleAccountHandler accountHandler = ((WindcentraleAccountHandler) bridge.getHandler());
+        return accountHandler == null ? null : accountHandler.getAPI();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == RefreshType.REFRESH) {
+            logger.debug("Refreshing {}", channelUID);
+            updateData();
+        } else {
+            logger.debug("This binding is a read-only binding and cannot handle commands");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Windcentrale handler '{}'", getThing().getUID());
+
+        WindmillConfiguration config = getConfig().as(WindmillConfiguration.class);
+        this.config = config;
+
+        Windmill windmill = Windmill.fromName(config.name);
+        this.windmill = windmill;
+
+        if (windmill == null) {
+            // only occurs when a mismatch is introduced between config parameter options and enum values
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Invalid windmill name: " + config.name);
```

Comment:
Consider providing an I18N text.

Reply:
I didn't do that as it will only occur during development. The thing won't be initialized if the windmill name is not in the list of config paramter values.